anospp_analysis/plasm.py

```diff
@@ -0,0 +1,679 @@
+               min_border=0, toolbar_location='below')
+    rects = Rect(x="x", y="recty",  width=1, height=1, fill_color="colors",
+                 line_color=None, fill_alpha=0.6)
+    p.add_glyph(source, rects)
+    p.yaxis.visible = False
+    p.grid.visible = False  
+
+    #sequence text view with ability to scroll along x axis
+    p1 = figure(title=None, plot_width=plot_width, plot_height=plot_height,
+                x_range=view_range, y_range=ids, tools="xpan,reset",
+                min_border=0, toolbar_location='below')#, lod_factor=1)          
+    glyph = Text(x="x", y="y", text="text", text_align='center',text_color="black",
+                text_font="monospace",text_font_size=fontsize)
+    rects = Rect(x="x", y="recty",  width=1, height=1, fill_color="colors",
+                line_color=None, fill_alpha=0.4)
+    p1.add_glyph(source, glyph)
+    p1.add_glyph(source, rects)
+
+    p1.grid.visible = False
+    p1.xaxis.major_label_text_font_style = "bold"
+    p1.yaxis.minor_tick_line_width = 0
+    p1.yaxis.major_tick_line_width = 0
+
+    p = gridplot([[p],[p1]], toolbar_location='below')
+    return p
+
+def get_colors(seqs):
+    """make colors for bases in sequence"""
+    text = [i for s in list(seqs) for i in s]
+    clrs =  {'a':'red','t':'green','g':'orange','c':'blue','-':'white', 'n':'black'}
+    colors = [clrs[i] for i in text]
+    return colors
+
+def uniques(xs):
+    return list(sorted(set(xi for x in xs for xi in x)))
+
+def avg(y):
+    return round(mean(y), 1)
+
+def plot_lims_plate(df, target, plate, fname, title=None, annot=True, cmap='coolwarm', center=None):
+    """
+    Plot a heatmap of the total read count for a given target on a plate.
+
+    Args:
+        df (pandas.DataFrame): A dataframe containing the read counts and positions on the plate.
+        target (str): The name of the target (e.g. 'P1', 'P2').
+        title (str): The title of the plot. Default is None.
+        plate (str): The name of the plate. Default is None.
+        annot (bool): Whether to annotate the heatmap cells with their values. Default is True.
+        cmap (str or colormap): The colormap to use. Default is 'coolwarm'.
+        center (float): The value at which to center the colormap. Default is None.
+        filename (str): The name of the file to save the plot to. Default is None.
+
+    Returns:
+        None
+    """
+    fig, ax = plt.subplots(1, 1, figsize=(14, 8))
+    col = f'total_reads_{target}'
+    sns.heatmap(df.pivot(index = 'lims_row', columns='lims_col', values= col), annot=annot, cmap=cmap, ax=ax, center=center)
+    ax.set_title(f"Total {target} reads for {plate}")
+    ax.hlines([i * 2 for i in range(9)], 0, 24, colors='k')
+    ax.vlines([j * 2 for j in range(13)], 0, 16, colors='k')
+    plt.tight_layout()
+    if fname is not None:
+        plt.savefig(fname, dpi=300, bbox_inches='tight')
+    plt.close(fig)
+
+def plot_plate_view(df, fname, P, title=None):
```

Comment:
- move to iplot.py
- if filename is None (default value) - return plot, if filename is string - save to file, else raise error
- iterate over P1 and P2 in this function to save in the same html
- pass ['P1','P2'] as `targets` argument (instead of P here) - use PLASM_TARGETS as a reference
- avoid using caps only variable names - this is the convention for constants